classes/class-wc-rest-connect-shipping-carrier-types-controller.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit();
+}
+
+if ( class_exists( 'WC_REST_Connect_Shipping_Carrier_Types_Controller' ) ) {
+	return;
+}
+
+/**
+ * Retrieve a list of carrier WooCommerce Shipping & Tax supports, along with the
+ * fields needed for each carrier in order to do carrier account registration.
+ */
+class WC_REST_Connect_Shipping_Carrier_Types_Controller extends WC_REST_Connect_Base_Controller {
+	/**
+	 * Carrier-types end point
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'connect/shipping/carrier-types';
+
+	/**
+	 * GET request
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get() {
+		$response = $this->api_client->get_carrier_types();
+		if ( is_wp_error( $response ) ) {
+			$this->logger->log( $response, __CLASS__ );
+			return $response;
+		}
+		return new WP_REST_Response(
+			[
+				'success' => true,
+				'fields'  => $response,
```

Comment:
Something I didn't pay attention to earlier - I am seeing the data returned in `fields > carriers`.

![Screen Shot 2020-11-02 at 3 48 25 PM](https://user-images.githubusercontent.com/273592/97923065-092df600-1d23-11eb-8739-1d5ef7d298ea.png)


Was there a reason why why having this structure? Should it just be `carriers => []`, rather than `fields => carriers => []`?

i.e.:
```suggestion
				'carriers'  => $response->carriers,
```
![Screen Shot 2020-11-02 at 3 52 05 PM](https://user-images.githubusercontent.com/273592/97923310-65911580-1d23-11eb-82a6-5c00603b2c66.png)

Reply:
> Was there a reason why why having this structure? 

I was thinking more of `success => true, data => {...}`. But `data` was too generic, so i chose `fields`. I like the suggestion. We don't need another layer to wrap the `carriers` object.

Reply:
Ah gotcha, that makes sense! Yeah, either `data => []` or `carriers => []` work for me! 👍

Reply:
Thanks, updated!

Reply:
Looks great, thanks for considering it!
![Screen Shot 2020-11-02 at 4 27 25 PM](https://user-images.githubusercontent.com/273592/97926177-552f6980-1d28-11eb-80a8-e14262cfe84d.png)